auth/api/iam/error.go

```diff
@@ -125,3 +126,16 @@ func (p oauth2ErrorWriter) Write(echoContext echo.Context, _ int, _ string, err
 	redirectURI.RawQuery = query.Encode()
 	return echoContext.Redirect(http.StatusFound, redirectURI.String())
 }
+
+const InvalidScope = ErrorCode("invalid_scope")
+
+var ErrInvalidScope = errors.New("invalid scope")
```

Comment:
could just return `OAuth2Error`? You're already unmarshalling it
Then you also have the description and stuff